tool/tctl/common/plugin/awsic_test.go

```diff
@@ -0,0 +1,196 @@
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package plugin
+
+import (
+	"math/rand/v2"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/tool/common"
+)
+
+func TestAWSICUserFilters(t *testing.T) {
+	t.Run("parse", func(t *testing.T) {
+		// GIVEN some arbitrary set of account name filter CLI args
+		cliArgs := awsICArgs{
+			userLabels: common.LabelFilters{
+				{"a": "alpha", "b": "bravo", "c": "charlie"},
+			},
+		}
+
+		// WHEN I attempt to convert the command-line args into filters
+		actualFilters, err := cliArgs.parseUserFilters()
+
+		// EXPECT the operation succeeds AND that the returned filters are an
+		// accurate representation of the command line args, in arbitrary order
+		require.NoError(t, err)
+		expectedFilters := []*types.AWSICUserSyncFilter{
+			{Labels: map[string]string{"a": "alpha", "b": "bravo", "c": "charlie"}},
+		}
+		require.ElementsMatch(t, expectedFilters, actualFilters)
+	})
+
+	t.Run("empty lists are valid", func(*testing.T) {
+		// GIVEN a cli arg set that doesn't specify any group filters
+		cliArgs := awsICArgs{}
+
+		// WHEN I attempt to convert the command-line args into filters
+		filters, err := cliArgs.parseUserFilters()
+
+		// EXPECT that the operation succeeds and returns an empty filter list
+		require.NoError(t, err)
+		require.Empty(t, filters)
+	})
+
+	t.Run("origin is applied", func(*testing.T) {
+		// GIVEN a CLI arg set that supplies user origin values
+		cliArgs := awsICArgs{
+			userLabels: common.LabelFilters{
+				{"a": "alpha", "b": "bravo", "c": "charlie"},
+			},
+			userOrigins: []string{
+				types.OriginEntraID,
+				types.OriginOkta,
+			},
+		}
+
+		// WHEN I attempt to convert the command-line args into filters
+		actualFilters, err := cliArgs.parseUserFilters()
+
+		// EXPECT that the operation succeeds and that the returned filters are
+		// an accurate representation of the command line args. Returned filters
+		// are not expected to have any particular order.
+		require.NoError(t, err)
+		expectedFilters := []*types.AWSICUserSyncFilter{
+			{Labels: map[string]string{"a": "alpha", "b": "bravo", "c": "charlie"}},
+			{Labels: map[string]string{types.OriginLabel: types.OriginOkta}},
+			{Labels: map[string]string{types.OriginLabel: types.OriginEntraID}},
+		}
+		require.ElementsMatch(t, expectedFilters, actualFilters)
+	})
+}
+
+func TestAWSICGroupFilters(t *testing.T) {
+	t.Run("parse", func(t *testing.T) {
+		// GIVEN some arbitrary set of account name filter CLI args
+		cliArgs := awsICArgs{
+			groupNameFilters: []string{"alpha", "bravo", "charlie"},
+		}
+
+		// WHEN I attempt to convert the command-line args into filters
+		actualFilters, err := cliArgs.parseGroupFilters()
+
+		// EXPECT the operation to succeed
```

Comment:
Maybe this is personal preference, but I would remove this comment. It's pretty clear that a line that says "require no error" is expecting that there wasn't an error.